x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/use_columns.tsx

```diff
@@ -106,6 +117,41 @@ const useEnabledColumn = ({ hasCRUDPermissions, startMlJobs }: ColumnsProps): Ta
   );
 };
 
+const useRuleSnoozeColumn = ({
+  hasCRUDPermissions,
+}: {
+  hasCRUDPermissions: boolean;
+}): TableColumn => {
+  const invalidateFetchRuleSnoozeSettings = useInvalidateFetchRulesSnoozeSettingsQuery();
+
+  return useMemo(
+    () => ({
+      field: 'snooze',
+      name: i18n.COLUMN_SNOOZE,
+      render: (_, rule: ExtendedRule) => {
+        if (rule.snoozeSettings instanceof Error) {
+          return (
+            <EuiToolTip content={rule.snoozeSettings.message}>
+              <EuiButtonIcon size="s" iconType="bellSlash" disabled />
+            </EuiToolTip>
+          );
+        }
+
+        return (
+          <RuleSnoozeBadge
+            snoozeSettings={rule.snoozeSettings}
+            hasCRUDPermissions={hasCRUDPermissions}
+            onChange={invalidateFetchRuleSnoozeSettings}
+          />
+        );
```

Comment:
I think this all should be encapsulated in the `RuleSnoozeBadge` component. It should use the `RulesTableContext` to get the state of the snooze settings and render itself according to it. `useInvalidateFetchRulesSnoozeSettingsQuery()` should be in it as well.